### PR TITLE
chore: Cut v2.10.0

### PR DESCRIPTION
**Pull request by mrueg**
**What this PR does / why we need it**:
```
## v2.10.0 / 2023-08-31

### Note
- By default experimental label and annotation metrics are disabled now to reduce memory usage. Before this change, they would include name and namespace labels.

* [BUGFIX] custommetrics: log no _info suffix in name only once per reading the configuration by @chrischdi in https://github.com/kubernetes/kube-state-metrics/pull/2157
* [BUGFIX] Don't crash on non-existent valueFrom path values by @chihshenghuang in https://github.com/kubernetes/kube-state-metrics/pull/2140
* [BUGFIX] Index out of range in metrics_store.SanitizeHeaders by @mrueg in https://github.com/kubernetes/kube-state-metrics/pull/2166
* [BUGFIX] custommetrics: always extract the headers but only write it when we have metrics by @chrischdi in https://github.com/kubernetes/kube-state-metrics/pull/2154
* [BUGFIX] Add filtering for Lease metrics by @ntoofu in https://github.com/kubernetes/kube-state-metrics/pull/2122
* [FEATURE] implement kube_pod_status_initialized_time by @opeco17 in https://github.com/kubernetes/kube-state-metrics/pull/2148
* [FEATURE] disable labels and annotations metrics when metric-annotations-… by @opeco17 in https://github.com/kubernetes/kube-state-metrics/pull/2145
* [FEATURE] Add webhooks client config service metrics by @dgrisonnet in https://github.com/kubernetes/kube-state-metrics/pull/2114
* [FEATURE] support kube_persistentvolumeclaim_deletion_timestamp by @maxime1907 in https://github.com/kubernetes/kube-state-metrics/pull/2074
* [FEATURE] support kube_persistentvolume_deletion_timestamp by @maxime1907 in https://github.com/kubernetes/kube-state-metrics/pull/2075
* [FEATURE] Adds new metric `kube_pod_service_account` by @swarup-stripe in https://github.com/kubernetes/kube-state-metrics/pull/2096
* [FEATURE] add volumemode to PVC info metric by @machadovilaca in https://github.com/kubernetes/kube-state-metrics/pull/2134
* [FEATURE] enable metric-annotations-allowlist and metric-labels-allowlist for ResourceQuota by @opeco17 in https://github.com/kubernetes/kube-state-metrics/pull/2175
* [FEATURE] allow field KV general matching by @rexagod in https://github.com/kubernetes/kube-state-metrics/pull/2067
* [FEATURE] hot reload kubeconfig by @opeco17 in https://github.com/kubernetes/kube-state-metrics/pull/2144
* [FEATURE] Add support for endpoint topology routing hints by @MarkSRobinson in https://github.com/kubernetes/kube-state-metrics/pull/2090

```
**How does this change affect the cardinality of KSM**: *(increases, decreases or does not change cardinality)*
-
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
-

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,25 @@
+## v2.10.0 / 2023-08-31
+
+### Note
+- By default experimental label and annotation metrics are disabled now to reduce memory usage. Before this change, they would include name and namespace labels.
+
+* [BUGFIX] custommetrics: log no _info suffix in name only once per reading the configuration by @chrischdi in https://github.com/kubernetes/kube-state-metrics/pull/2157
+* [BUGFIX] Don't crash on non-existent valueFrom path values by @chihshenghuang in https://github.com/kubernetes/kube-state-metrics/pull/2140
+* [BUGFIX] Index out of range in metrics_store.SanitizeHeaders by @mrueg in https://github.com/kubernetes/kube-state-metrics/pull/2166
+* [BUGFIX] custommetrics: always extract the headers but only write it when we have metrics by @chrischdi in https://github.com/kubernetes/kube-state-metrics/pull/2154
```
Comment (dgrisonnet) [inline CHANGELOG.md:9]: These are all related to custom metrics so it feels inconsistent to have some prefixed by `custommetrics:` and some not

Comment (mrueg) [reply CHANGELOG.md:9]: I was using the PR title as generated by github when drafting a release.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,25 @@
+## v2.10.0 / 2023-08-31
+
+### Note
+- By default experimental label and annotation metrics are disabled now to reduce memory usage. Before this change, they would include name and namespace labels.
```
[the remark this answers is not in the document]
Comment (mrueg) [reply CHANGELOG.md]: Thanks, applied!

File: CHANGELOG.md (hunk 1 above)
Comment (mrueg) [reply CHANGELOG.md]: I updated the titles a bit.

Comment (dgrisonnet) [reply CHANGELOG.md]: ok

**Pull request merged**